bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/HomeWizardHandler.java

```diff
@@ -0,0 +1,200 @@
+
+import java.io.IOException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.homewizard.data.P1Payload;
+import org.openhab.core.io.net.http.HttpUtil;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link HomeWizardHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Daniël van Os - Initial contribution
+ */
+@NonNullByDefault
+public class HomeWizardHandler extends BaseThingHandler {
+
+    private HomeWizardConfiguration config = new HomeWizardConfiguration();
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private final Gson gson = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
+            .create();
+
+    private String apiURL = "";
+    private String meterModel = "";
+    private int meterVersion = 0;
+
+    /**
+     * Constructor
+     *
+     * @param thing The thing to handle
+     */
+    public HomeWizardHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Not listening to any commands.
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    /**
+     * If a host has been specified start polling it
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(HomeWizardConfiguration.class);
+        if (configure()) {
+            pollingJob = scheduler.scheduleWithFixedDelay(this::pollingCode, 0, config.refreshDelay, TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * Check the current configuration
+     *
+     * @return true if the configuration is ok to start polling, false otherwise
+     */
+    private boolean configure() {
+        if (config.ipAddress.trim().isEmpty()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Missing ipAddress/host configuration");
+            return false;
+        } else {
+            updateStatus(ThingStatus.UNKNOWN);
+            apiURL = String.format("http://%s/api/v1/data", config.ipAddress.trim());
+            return true;
+        }
+    }
+
+    /**
+     * dispose: stop the poller
+     */
+    @Override
+    public void dispose() {
+        var job = pollingJob;
+        if (job != null && !job.isCancelled()) {

```

Comment:
No harm in canceling a task that is already cancelled.
```suggestion
        if (job != null) {

```